### ISSUE TITLE
Plan Wizard - Step 2: Client side validation

### ISSUE THREAD

**Issue by priley86** [priley86/miq_v2v_ui_plugin#206]
I was not able to implement this feature this sprint and I think we should carry it forward. See comments in CSVDropzoneField.js for how we might implement this:

```
          // TODO: client side validation logic should occur here and should prevent calling
          // onCSVParseSuccess until all validation is done...
          // i.e. setState with errors and show them here instead.

          /**
           * Need to check the following is done:
           * 1. csv is empty
           * 2. csv contains malformed rows (maybe print them out?)
           * 3. Ask user to import the csv again
           */
```

**Comment by vconzola** [comment]
We decided in Tuesday's sprint planning that we will test the .csv file for three conditions:
- file is the correct format
- file contains a blank row
- file contains a missing VM name

For the first condition we should return an empty state with an error:
![9c1-createplan-vms-import empty state 4 step - wrong format](https://user-images.githubusercontent.com/22823035/38955027-60fdf7ce-4321-11e8-9deb-93e9ae750794.png)

For the second and third conditions we should display the file contents and highlight the row(s) with missing data:
![9c3-createplan-vms-import selection table w bad data](https://user-images.githubusercontent.com/22823035/38955098-9a9c7546-4321-11e8-8088-c638c6ecb147.png)

**Comment by AparnaKarve** [comment]
I can work on this while I'm working on the other CSV requirements from Bill.

**Comment by AparnaKarve** [comment]
@vconzola For the first condition, it is possible to display a reason, as to why a .csv is not in the expected format.
For example, it is possible to display the reason that the number of columns are inconsistent on line2
(this is an error message displayed as-is from the csv parser)

<img width="809" alt="screen shot 2018-04-19 at 8 42 49 am" src="https://user-images.githubusercontent.com/1538216/39002569-afc15e78-43ad-11e8-87b7-1d893d14af47.png">

So, is it OK to display the error details as shown above?

**Comment by vconzola** [comment]
@AparnaKarve If an error only pertains to the formatting of a single row or a few rows, I'd like to display the contents of the file as in the second mockup above. The tooltip on the error icon can then provide details about the error. The rows with errors are disabled so they can't be selected. The only time we should display the error empty state (first mockup) is when the file is so buggered up (wrong type, missing required columns, etc.) that it can't be opened or displayed properly in the table. 

One of the design directions we were given by Marco (product owner) is that we should not block the user's progress unless absolutely necessary. Let me know if you want to discuss.

**Comment by AparnaKarve** [comment]
> The only time we should display the error empty state (first mockup) is when the file is so buggered up (wrong type, missing required columns, etc.) that it can't be opened or displayed properly in the table.

@vconzola The error above - "Number of columns is inconsistent on line 2" falls in the above category.
The CSV parser fails to parse because of the column inconsistency, and as a result, we can't get to the CSV table screen.

If we prefer not to show the error details in the first mockup, that is fine.
(I just wanted to mention that the error information is available if needed.)

**Comment by vconzola** [comment]
@AparnaKarve OK, thanks for the clarification. In that case, what you're showing is fine. Just make sure the font type and size are consistent for the whole message text.

**Comment by serenamarie125** [comment]
Fixed by @AparnaKarve with #239
